server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -327,14 +327,14 @@ private void checkNestedLimit(long limit) {
         }
     }
 
-    public boolean hasNested() {
-        return hasNested;
-    }
-
     public Map<String, ObjectMapper> objectMappers() {
         return objectMappers;
     }
 
+    public NestedLookup nestedLookup() {
+        return nestedLookup;
```

Comment:
if we left hasNested like I proposed above, would it make sense to enforce that this is only called when hasNested returns true, or would the NestedLookup also retrieved in the case of NestedLookup.EMPTY?

Reply:
I think the answer to this is that NestedLookup would be retrieved also when it returns EMPTY. Which is fine.